iohub/ngff_utils.py

```diff
@@ -0,0 +1,341 @@
+    func,
+    position: Position,
+    output_path: Path,
+    input_channel_indices: list[int],
+    output_channel_indices: list[int],
+    t_idx_in: int,
+    t_idx_out: int,
+    **kwargs,
+) -> None:
+    """
+    Load a CZYX array from a Position object, apply a transformation to CZYX.
+
+    Parameters
+    ----------
+    func : CZYX -> CZYX function
+        The function to be applied to the data
+    position : Position
+        The position object to read from
+    output_path : Path
+        The path to output OME-Zarr Store
+    input_channel_indices : list
+        The channel indices to process.
+        If empty list, process all channels.
+        Must match output_channel_indices if not empty
+    output_channel_indices : list
+        The channel indices to write to.
+        If empty list, write to all channels.
+        Must match input_channel_indices if not empty
+    t_idx_in : int
+        The time index to process
+    t_idx_out : int
+        The time index to write to
+    kwargs : dict
+        Additional arguments to pass to the CZYX function
+    """
+
+    # TODO: temporary fix to slumkit issue
+    if _is_nested(input_channel_indices):
+        input_channel_indices = [
+            int(x) for x in input_channel_indices if x.isdigit()
+        ]
+    if _is_nested(output_channel_indices):
+        output_channel_indices = [
+            int(x) for x in output_channel_indices if x.isdigit()
+        ]
+
+    # Check if t_idx_in should be added to the func kwargs
+    # This is needed when a different processing is needed for each time point, for example during stabilization
+    all_func_params = inspect.signature(func).parameters.keys()
+    if "t_idx_in" in all_func_params:
+        kwargs["t_idx_in"] = t_idx_in
+
+    # Process CZYX given with the given indeces
+    # if input_channel_indices is not None and len(input_channel_indices) > 0:
+    click.echo(f"Processing t={t_idx_in} and channels {input_channel_indices}")
+    czyx_data = position.data.oindex[t_idx_in, input_channel_indices]
+    if not _check_nan_n_zeros(czyx_data):
+        transformed_czyx = func(czyx_data, **kwargs)
+        # Write to file
+        with open_ome_zarr(output_path, mode="r+") as output_dataset:
+            output_dataset[0].oindex[
+                t_idx_out, output_channel_indices
+            ] = transformed_czyx
+        click.echo(
+            f"Finished Writing.. t={t_idx_in} and channel output={output_channel_indices}"
+        )
+    else:
+        click.echo(f"Skipping t={t_idx_in} due to all zeros or nans")
+
+
+# TODO: modify how we get the time and channesl like recOrder (isinstance(input, list) or instance(input,int) or all)
+def process_single_position(
+    func,
+    input_data_path: Path,
+    output_path: Path,
+    time_indices_in: list = "all",
+    time_indices_out: list = [],
+    input_channel_idx: list = [],
+    output_channel_idx: list = [],
+    num_processes: int = mp.cpu_count(),
```

Comment:
```suggestion
    num_processes: int = 1
```

`mp.cpu_count` can cause issues in the HPC if you allocate fewer cpus then the total cpus in the node -- which is often the case.